debugging/helper-cleanout-server.js

```diff
@@ -109,19 +116,33 @@ exports.run = function (extra, args, cleanout) {
       }
     }
 
+    const t2 = new Date();
+    print("INFO " + (t2));
+
     if (cleanout) {
-      print("INFO shards to be moved away from node " + shortName + ": " + count);
+      print("INFO " + task + ": shards to be moved away from node " + shortName + ": " + count);
       if (count === 0) break;
     } else {
-      print("INFO shards to be moved away from node " + shortName + ": " + count + ", leader-only: " + leaderOnly);
+      print("INFO " + task + ": shards to be moved away from node " + shortName + ": " + count +
+            ", leader-only: " + leaderOnly);
       if (count === leaderOnly) break;
     }
-    internal.wait(sleep);
+
+    const d = (t2 - t1) / 1000;
+    t1 = t2;
+
+    if (d < sleep) {
+      internal.wait(sleep - d);
+    }
   } while (count > 0);
 
   while (res.status === 'ToDo') {
     res = helper.httpWrapper('GET', '/_admin/cluster/queryAgencyJob?id=' + jobId);
+
+    const t2 = new Date();
```

Comment:
Is t2 used at all besides Print?
If we want to keep the print this is good.
If not, we need to remove t2 here as well.

Reply:
Yes, it can take upto an hour. So t2 is showing the current time to ensure that the script is not stuck